### PR TITLE
Remove dependence on `using namespace`: Build of 'common' through osdc/Objecter.cc

### PR DESCRIPTION
**Pull request by adamemerson**
Go through a chunk of stuff removing the need for 'using namespace std' and 'using namespace ceph'.

Partial, of course, but getting Objecter.cc to compile means that the most commonly used headers used by targets in common should be cleaned up. Others are free to continue with further installments, if they wish.

### CODE REVIEW

**Comment by tchaikov** [comment]
retest this please.

File: src/include/encoding.h [hunk 1]
```diff
@@ -1439,7 +1439,7 @@ decode(std::array<T, N>& v, bufferlist::const_iterator& p)
   } while (false);							\
   if (struct_end) {							\
     if (bl.get_off() > struct_end)					\
-      throw buffer::malformed_input(DECODE_ERR_PAST(__PRETTY_FUNCTION__)); \
+      throw ::ceph::buffer::malformed_input(DECODE_ERR_PAST(__PRETTY_FUNCTION__)); \
```
Comment (tchaikov) [inline src/include/encoding.h]: i wonder, if `buffer` is already in the same naming space of `decode()`, shall we guard `buffer` with `::ceph::`? i read libstdc++, libc++ and boost . none of them are referencing the class with the fully-qualified namespace prefix. what's your consideration of this change here?

File: src/include/object.h [hunk 2]
```diff
@@ -15,11 +15,12 @@
 #ifndef CEPH_OBJECT_H
 #define CEPH_OBJECT_H
 
-#include <stdint.h>
-#include <stdio.h>
-
+#include <cstdint>
+#include <cstdio>
```
Comment (tchaikov) [inline src/include/object.h]: :+1:

File: src/auth/AuthServiceHandler.h [hunk 3]
```diff
@@ -18,7 +18,7 @@
 #include <stddef.h>              // for NULL
 #include <stdint.h>              // for uint64_t
 #include "common/entity_name.h"  // for EntityName
-#include "include/buffer_fwd.h"  // for bufferlist
+#include "include/buffer_fwd.h"  // for ceph::buffer::list
```
[the remark this answers is not in the document]
Comment (adamemerson) [reply src/auth/AuthServiceHandler.h]: It declares both. It declares the buffer::list class and adds a using for bufferlist.

File: src/include/encoding.h (hunk 1 above)
Comment (adamemerson) [reply src/include/encoding.h]: It's a macro. The body could be expanded in any scope, so I want to fully qualify it.

File: src/auth/AuthServiceHandler.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (tchaikov) [reply src/auth/AuthServiceHandler.h]: ahh, sorry! i missed "inline"!

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2019-04-01_10:38:29-rados-wip-kefu-testing-2019-04-01-1531-distro-basic-mira/

**Comment by sebastian-philipp** [comment]
Is https://jenkins.ceph.com/job/ceph-pull-requests/560/console related? 

```
Failed to load class: cas (/home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libcls_cas.so): /home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libcls_cas.so: undefined symbol: cls_cxx_chunk_write_and_set(void*, int, int, ceph::buffer::v14_2_0::list*, unsigned int, ceph::buffer::v14_2_0::list*, int)
Failed to load class: cephfs (/home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libcls_cephfs.so): /home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libcls_cephfs.so: undefined symbol: typeinfo for PGLSFilter
Failed to load class: hello (/home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libcls_hello.so): /home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libcls_hello.so: undefined symbol: typeinfo for PGLSFilter
Failed to load class: log (/home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libcls_log.so): /home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libcls_log.so: undefined symbol: cls_cxx_map_read_header(void*, ceph::buffer::v14_2_0::list*)
Failed to load class: rgw (/home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libcls_rgw.so): /home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libcls_rgw.so: undefined symbol: cls_cxx_map_read_header(void*, ceph::buffer::v14_2_0::list*)
Failed to load class: user (/home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libcls_user.so): /home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libcls_user.so: undefined symbol: cls_cxx_map_read_header(void*, ceph::buffer::v14_2_0::list*)
Failed to load class: version (/home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libcls_version.so): /home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libcls_version.so: undefined symbol: cls_gen_rand_base64(char*, int)
```
?

**Comment by tchaikov** [comment]
@sebastian-philipp i don't think it's related. i don't even think "Failed to load class" fails the test(s). the reason we have these failures is that `src/test/librados_test_stub/LibradosTestStub.cc` only offers the minimum set of API used by class objects to support librbd related test.

**Comment by tchaikov** [comment]
#28250 is posted to address these annoying warnings.

**Pull request merged**